src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -592,6 +592,42 @@ void ClusterResourceScheduler::DeleteResource(const std::string &node_id_string,
   }
 }
 
+std::string ClusterResourceScheduler::SerializedTaskResourceInstances(
+    std::shared_ptr<TaskResourceInstances> task_allocation) const {
```

Comment:
I don't think it's a good idea to serialize resources to json string. It's not able to reuse the cached workers for the same resource set with different string, like "1c2g" and "2c1g".

How about to use a data structure such as `Tags` for this.

Reply:
@jovany-wang I thins `Tags` is great. How about working for it in another PR?